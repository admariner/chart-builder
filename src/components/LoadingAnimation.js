// @flow
import React, { Component } from 'react'
import cx from 'classnames'
import classes from './LoadingAnimation.module.css'
import type { Node } from 'react'

export default class LoadingAnimation extends Component<{
  hideOverlay?: ?boolean,
  label?: Node
}> {
  render() {
    const { hideOverlay, label } = this.props
    return (
      <div
        className={cx(classes.container, hideOverlay && classes.hideOverlay)}
      >
        <div className={classes.center}>
          <div className={classes.loader}>Loading...</div>
          {label && <small>{label}</small>}
        </div>
      </div>
    )
  }
}
