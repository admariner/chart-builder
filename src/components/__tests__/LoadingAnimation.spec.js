import React from 'react'
import LoadingAnimation from '../LoadingAnimation'

it('renders', () => {
  snap(<LoadingAnimation />)
})

it('renders with a label', () => {
  snap(<LoadingAnimation label="foo" />)
})

it('renders without overlay', () => {
  snap(<LoadingAnimation hideOverlay />)
})
