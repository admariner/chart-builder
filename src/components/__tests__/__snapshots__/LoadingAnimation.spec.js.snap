// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<div
  className="container"
>
  <div
    className="center"
  >
    <div
      className="loader"
    >
      Loading...
    </div>
  </div>
</div>
`;

exports[`renders with a label 1`] = `
<div
  className="container"
>
  <div
    className="center"
  >
    <div
      className="loader"
    >
      Loading...
    </div>
    <small>
      foo
    </small>
  </div>
</div>
`;

exports[`renders without overlay 1`] = `
<div
  className="container hideOverlay"
>
  <div
    className="center"
  >
    <div
      className="loader"
    >
      Loading...
    </div>
  </div>
</div>
`;
